building-and-packaging/material/packaging_debian_and_spack_exercise.md

```diff
@@ -0,0 +1,99 @@
+# Packages with CPack and Spack
+
+This exercise is about packaging code with CPack and Spack. In this exercise we use adapted versions of the code from the previous week's exercise and create packages for it. The exercise is split into two parts: 1. CPack and 2. Spack. There are different GitHub repositories for the two subtasks.
+
+At the end of the exercise you find a section with hints and remarks. Make sure to check this section.
+
+Deadline: **Thursday, December 7th, 2021, 9:00**
+
+## 1. CPack
+
+**Attention/TODO**: Use [test repository](https://github.com/Simulation-Software-Engineering/cpack-exercise-test) for testing the exercise.
+
+The ultimate goals is to have a CMake/CPack configuration that allows us to generate packages for different targets. In this exercise we want to create a `tar.gz` package and a Debian `deb` package. We want to package this [HelloWorld example from GitHub](https://github.com/Simulation-Software-Engineering/cpack-exercise).
+
+When setting up the package make sure that the executable `cpackexample` and also the library `cpackexamplelib` are build and installed in appropriate locations. You should also ensure that the header files for the library are installed properly. Please package `cpackexamplelib` as **shared** library. The package should contain enough information (at least maintainer, contact address, project description, vendor, version number, homepage). If you do not want to use your actual name/email address, please use your GitHub username and make up an email address).
+
+For the Debian package, make sure that
+
+- the executable is stripped.
+- the list of dependencies if generated for the Debian package.
+- the Debian package is generated with default file naming scheme.
+
+Please use a suitable Docker container for this task. You should be able to reuse the Docker container from last week's exercise. The Docker container will be especially important if you want to work on the optional task
+
+### Workflow (CPack)
+
+- Create a fork of the [GitHub repository](https://github.com/Simulation-Software-Engineering/cpack-exercise).
+- Adapt and create the files necessary to create packages via CPack.
+- Also add your `Dockerfile` to the repository.
+- Submit your code via a pull request. You can also open a (draft) pull request before you are incorporated all changes. In this case, do not forget to mark the pull request as ready to review when you are finished.
+    - The pull request should be named `[USERNAME] CPack exercise`, e.g., `[jaustar] CPack exercise`.
```

Comment:
```suggestion
    - The pull request should be named `[USERNAME] CPack exercise`, e.g., `[ajaust] CPack exercise`.
```
On GitHub the GitHub user name?

Reply:
I normally prefer GitLab usernames, but if we used the GitHub username for submissions so far I am fine with that as well.